lib/Classifiers/Classifier.php

```diff
@@ -173,9 +173,13 @@ private function getConvertedFilePath(Node $file): string {
 		$tmpname = $this->tempManager->getTemporaryFile('.jpg');
 
 		try {
-			// Downscale into a temporary JPEG file
+			// Convert to a temporary JPEG file optionally downscaling
 			$imagick = new \Imagick($path);
-			$imagick->scaleImage(4096, 4096, true);
+			$dimensions = $imagick->getImageGeometry();
+			if ($dimensions['width'] > 4096 || $dimensions['height'] > 4096) {
```

Comment:
Good catch. Imagick before 3.0.0 used to leave the image untouched if it was smaller than the specified scaling; I didn't know the behavior had changed.